assets/semgrep_rules/client/unsafejs-in-cpp.yaml

```diff
@@ -0,0 +1,33 @@
+rules:
+  - id: unsafe-js-in-cpp-strings
+    metadata:
+      author: Andrea Brancaleoni <abc@pompel.me>
+      confidence: LOW
+      assignees: |
+        diracdeltas
+        thypon
+      source: https://github.com/brave/security-action/blob/main/assets/semgrep_rules/client/unsafejs-in-cpp.yaml
+    message: Unsafe JS in CPP strings
+    languages:
+      - c
+      - cpp
```

Comment:
is there already a rule covering innerhtml/document.write in HTML/JS files?

Reply:
https://semgrep.dev/playground/r/7ZTYRP/javascript.browser.security.insecure-document-method.insecure-document-method if the innerHTML or document.write is a variable, yup

Reply:
not if it's a string